code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -740,6 +740,12 @@ emp_act
 	. = ..()
 	dna.species.water_act(src, volume, temperature, source, method)
 
+/mob/living/carbon/human/attackby(obj/item/I, mob/user, params)
+	if((user.a_intent in list(INTENT_DISARM, INTENT_HARM)) && try_protection(user))
```

Comment:
It feels a little snowflakey to me to have revcode be the only reason for this existing, do you think you could do this with signals instead? Register them on status application, and unregister when the status effect expires. From the looks of things, you might need to tweak the way `COMSIG_PARENT_ATTACKBY` is handled (making it respond properly to `COMPONENT_CANCEL_ATTACK_CHAIN`).

Reply:
Attackchain code is a mess... I wasn't able to really alter it to use COMPONENT_CANCEL_ATTACK_CHAIN without being a bunch of garbage and intruding in SO many files. I'm gonna be adding a new signal instead and see if that might be better

Reply:
I made it use a special signal on humans instead, since the rest of attackby code would need to be updated to accommodate a 3rd new return value (which is a lot of code...). Works the same as before, but should be a bit cleaner now